<launch>

<arg name="rviz" default="true" />
<param name="use_sim_time" value="true" />

<!--    <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch" />-->

<!--    <include file="$(find receive_xsens)/launch/receive_xsens.launch" />-->

<node pkg="location" type="locationSyncwithRostime" name="locationSyncwithRostime"  output="screen">
  <param name="mapPCDFileName" value="$(find location)/map/yuquan.pcd"/>
  <param name="lidarName" value="velodyne_packets"/>
  <param name="enableSwitchLidarData" value="false"/>
  <param name="switchBufferSize" value="4"/>
  <param name="enablePublishGlobalDenseMap" value="false"/>

</node>

<node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0.0 0 0.85 0 0 0 base_link velodyne 50" />
<node pkg="tf" type="static_transform_publisher" name="link3_broadcaster" args="0.0 0 0.8 0 0 0 base_link imu 50" />
<node pkg="tf" type="static_transform_publisher" name="link4_broadcaster" args="0.0 0 0.0 0 0 0 map odom 50" />
<node pkg="tf" type="static_transform_publisher" name="link5_broadcaster" args="0.0 0.0 -0.1 -1.57 -0.01 -1.9 velodyne camera_link 50" />

<node name="MapServer" pkg="map_server" type="map_server" output="screen" args="$(find location)/map/yuquan.yaml"/>

<node pkg="detection" type="detection" name="detection" output="screen">
    <param name="cameraFrame" value="camera_link"/>
    <param name="cameraPoint" value="/camera/depth/color/points"/>
    <param name="odomFrame" value="odom"/>
    <param name="FilterFieldName" value="z"/>
    <param name="setFilterLimitsUp" value="5.0"/>
    <param name="setFilterLimitsDown" value="0.0"/>
    <param name="use_static_tf" value="false"/>
    <param name="elevation_wait_time" value="10"/>
</node>

<node pkg="detection" type="rawmapping" name="rawmapping" output="screen">
    <param name="cameraFrame" value="camera_link"/>
    <param name="mpaFrame" value="odom"/>
    <param name="cameraPoint" value="/camera/depth/color/points"/>
    <param name="FilterFieldName" value="z"/>
    <param name="setFilterLimitsUp" value="2.0"/>
    <param name="setFilterLimitsDown" value="0.3"/>
    <param name="setFilterheight" value="0.15"/>
    <param name="setOutlierFilter" value="false"/>
    <param name="setOutlierFilterMeanK" value="10"/>
    <param name="setOutlierFilterThresh" value="1.0"/>
    <param name="voxelsize" value="0.04"/>
    <param name="roshz" value="1000"/>
    <param name="wait_surfel_finished" value="10000"/>
    <param name="setPointNumLimit" value="250000"/>
</node>

<group if="$(arg rviz)">
  <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find location)/rviz_cfg/ndt_matching.rviz" />
</group>

</launch>
